internal/cluster/transition.go

```diff
@@ -347,6 +353,52 @@ func (th *transitionHandler) PostRefreshCluster(reason string) stateswitch.PostT
 	return ret
 }
 
+func (th *transitionHandler) PostRefreshLogsProgress(progress string) stateswitch.PostTransition {
+	return func(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) error {
+		sCluster, _ := sw.(*stateCluster)
+		return updateLogsProgress(th.log, th.db, sCluster.cluster, swag.StringValue(sCluster.cluster.Status), progress)
+	}
+}
+
+//check if log collection on cluster level reached timeout
+func (th *transitionHandler) IsLogCollectionTimedOut(sw stateswitch.StateSwitch, args stateswitch.TransitionArgs) (bool, error) {
```

Comment:
What will this timeout do?
If we are collecting logs isn't the cluster already in `error` state?

Reply:
It will move the logs_info to timeout so we stop waiting for logs.